__tests__/app.test.js

```diff
@@ -373,6 +373,42 @@ describe("GET /api/articles?sort_by=:column&order=:order / Responds with an arra
   });
 });
 
+describe("GET /api/articles?topic=:topic / Responds with an array of articles filtered by topic", () => {
+  test("200: Responds with an array of articles filtered by cats topic", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(1);
+        articles.forEach((article) => {
+          expect(article.topic).toBe("cats");
+        });
+      });
+  });
+  test("200: Responds with an array of articles filtered by mitch topic AND in descending order by article_id", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch&sort_by=article_id&order=desc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(12);
+        expect(articles).toBeSortedBy("article_id", {
+          descending: true,
+        });
+        articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("404: Responds with an error message when given a valid but non-existent topic query", () => {
+    return request(app)
+      .get("/api/articles?topic=idontexistatall")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not found");
+      });
+  });
+});
```

Comment:
Good work! 

There is a topic that exists but doesn't have any articles attached to it yet - we should get a 200 and an empty array in that case!